modin/pandas/test/dataframe/test_join_sort.py

```diff
@@ -306,82 +309,68 @@ def test_merge(test_data, test_data2):
         modin_df.merge("Non-valid type")
 
 
-@pytest.mark.parametrize("data", test_data_values, ids=test_data_keys)
-@pytest.mark.parametrize("axis", axis_values, ids=axis_keys)
+@pytest.mark.parametrize("axis", [0, 1])
 @pytest.mark.parametrize(
     "ascending", bool_arg_values, ids=arg_keys("ascending", bool_arg_keys)
 )
 @pytest.mark.parametrize("na_position", ["first", "last"], ids=["first", "last"])
-@pytest.mark.parametrize(
-    "sort_remaining", bool_arg_values, ids=arg_keys("sort_remaining", bool_arg_keys)
-)
-def test_sort_index(data, axis, ascending, na_position, sort_remaining):
-    modin_df = pd.DataFrame(data)
-    pandas_df = pandas.DataFrame(data)
+def test_sort_index(axis, ascending, na_position):
+    data = test_data["float_nan_data"]
+    modin_df, pandas_df = pd.DataFrame(data), pandas.DataFrame(data)
 
     # Change index value so sorting will actually make a difference
-    if axis == "rows" or axis == 0:
+    if axis == 0:
         length = len(modin_df.index)
-        modin_df.index = [(i - length / 2) % length for i in range(length)]
-        pandas_df.index = [(i - length / 2) % length for i in range(length)]
+        for df in [modin_df, pandas_df]:
+            df.index = [(i - length / 2) % length for i in range(length)]
+
     # Add NaNs to sorted index
-    if axis == "rows" or axis == 0:
-        length = len(modin_df.index)
-        modin_df.index = [
-            np.nan if i % 2 == 0 else modin_df.index[i] for i in range(length)
-        ]
-        pandas_df.index = [
-            np.nan if i % 2 == 0 else pandas_df.index[i] for i in range(length)
-        ]
-    else:
-        length = len(modin_df.columns)
-        modin_df.columns = [
-            np.nan if i % 2 == 0 else modin_df.columns[i] for i in range(length)
-        ]
-        pandas_df.columns = [
-            np.nan if i % 2 == 0 else pandas_df.columns[i] for i in range(length)
-        ]
-
-    modin_result = modin_df.sort_index(
-        axis=axis, ascending=ascending, na_position=na_position, inplace=False
-    )
-    pandas_result = pandas_df.sort_index(
-        axis=axis, ascending=ascending, na_position=na_position, inplace=False
-    )
-    df_equals(modin_result, pandas_result)
+    meta = {0: "index", 1: "columns"}
+    for df in [modin_df, pandas_df]:
+        sort_index = getattr(df, meta[axis])
+        setattr(
+            df,
+            meta[axis],
+            [np.nan if i % 2 == 0 else sort_index[i] for i in range(len(sort_index))],
+        )
```

Comment:
will it be the same? Current approach is a bit confusing
```suggestion
    for df in [modin_df, pandas_df]:
        sort_index = df.axes[axis]
        df.set_axis(
            [np.nan if i % 2 == 0 else sort_index[i] for i in range(len(sort_index))],
            axis=axis,
            inplace=True
         )
        
```